### PR TITLE
fix: DM card fixes, tests, docs, and config cleanup

### LINKED ISSUE #4982: Sherman and Token
My opponent play General Sherman when I have token (I played ToC) on board, one of them is Fangtooth Cavern.
When I kill Sherman, TCO allow me to put Fangtooth Cavern to play as artifact.
I believe it's a bug.

### LINKED ISSUE #5006: Monster zero deploy issue
Depending on which half of Monster Zero is being played, "deploy" keyword is not being resolved.
Looking at DM.json data, I believe this is because the second half of this gigantic creature is missing its "deploy" keyword

### PR DESCRIPTION
**Pull request by dmikalova**
## Summary

- Fix `PutIntoPlayAction` to handle gigantic creature play allowances correctly
- Fix card implementations: GeneralSherman, Bastionclaw, ExeldonYash, GiltspineMesmerist
- Add tests for: GeneralSherman, MemroxTheRed, GiltspineMesmerist, Bastionclaw, ExeldonYash, Hydrogan, MonsterZero, ShadowGloomcoil
- Document `autoResolve` option and 'If you do' clause patterns in `docs/card-abilities.md`
- Update `test/AGENTS.md` with testing guidance
- Add `config/test.json5` to silence node-config warning during test runs
- Fixes #4982 and fixes #5006

---

> [!NOTE]
> **Medium Risk**
> Touches core game-flow logic for forced triggered ability ordering and `putIntoPlay` deploy handling (including gigantic composition), which can subtly affect many cards. Broad new/updated tests reduce risk but regressions could surface in edge-case trigger windows.
> 
> **Overview**
> Fixes several card and engine edge cases around *forced trigger ordering* and *put-into-play behavior*. `ForcedTriggeredAbilityWindow` now only auto-resolves when **all** queued triggers are `autoResolve`, and `PutIntoPlayAction` now treats gigantic creatures’ shared printed keywords (e.g. `deploy`) correctly before composition.
> 
> Updates multiple card implementations to match intended rules: `GeneralSherman` only returns purged **creatures** to play, `ExeldonYash` gates the “if you do” payoff on actual `onCapture` events, `HarmonicPack` only applies the extra 3 damage when the initial 2 damage was fully applied, `Cosmicrux` is simplified to a direct reaction damage action, `Bastionclaw` key-cost modifier applies to the current controller, and `GiltspineMesmerist` marks its ready-trigger as `autoResolve`.
> 
> Adds extensive regression tests for these interactions (including animated artifacts, token creatures, armor/ward/invulnerable, and complex multi-trigger windows), expands docs for `autoResolve` and chained “If you do” patterns, adds `config/test.json5` to silence node-config test warnings, and updates the dev auto-deploy workflow to track `master`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ce0b97901aa8766562912fb92074da84b1fff072. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: config/test.json5 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Empty config for NODE_ENV=test. Tests use the default config; this file
```
Comment (dmikalova) [inline config/test.json5:1]: This fixes a minor config issue in the tests

File: server/game/cards/05-DT/GeneralSherman.js [hunk 2]
```diff
@@ -22,7 +22,9 @@ class GeneralSherman extends Card {
                         onCardLeavesPlay: (event, context) => event.card === context.source
                     },
                     gameAction: ability.actions.sequentialPutIntoPlay((context) => ({
-                        forEach: context.event.clone.clonedPurgedCards
+                        forEach: context.event.clone.clonedPurgedCards.filter(
```
Comment (dmikalova) [inline server/game/cards/05-DT/GeneralSherman.js:25]: Added this fix in while validating hydrogan

File: server/game/cards/08-AS/HarmonicPack.js [hunk 3]
```diff
@@ -20,34 +20,52 @@ class HarmonicPack extends Card {
...
-                        }))
-                    ]),
-                    message: '{0} uses {1} to {3}discard {4}',
-                    messageArgs: (context) => [
-                        preThenContext.targets.creature ? 'deal 3 more damage and ' : '',
-                        context.preThenEvent.card
-                    ]
-                }
-            })
+            then: (preThenContext) => {
```
Comment (dmikalova) [inline server/game/cards/08-AS/HarmonicPack.js:23]: It came up with exeldon yash that "if you do" depends on everything before it being true. So HP is updated that you must deal 2 damage (armor/ward/invulnerable prevent this) and reveal a card from archive.

File: server/game/cards/07-GR/GiltspineMesmerist.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ class GiltspineMesmerist extends Card {
             when: {
                 onCardReadied: (event) => event.exhausted
             },
+            autoResolve: true,
```
Comment (dmikalova) [inline server/game/cards/07-GR/GiltspineMesmerist.js:12]: Giltspine and Cosmicrux are in the same timing window, and giltspine can autoresolve if other abilities are not resolving.

File: server/game/cards/11-VM25/Cosmicrux.js [hunk 5]
```diff
@@ -3,18 +3,13 @@ const Card = require('../../Card.js');
 class Cosmicrux extends Card {
     // After a player readies a creature, deal 1 to it.
     setupCardAbilities(ability) {
-        this.interrupt({
+        this.reaction({
```
Comment (dmikalova) [inline server/game/cards/11-VM25/Cosmicrux.js:6]: Updated again with new understanding. It is still the case that all creatures ready simultaneously, however the way cosmicrux is worded (and giltspine mesmerist), it causes a trigger for each readied creature to deal 1 damage, rather than a sack of coins style 1 trigger to deal 1 damage to each readied creature.

The implications of this:
- if the target creature dies from cosmicrux, giltspine still discards for that ready
- if the source (cosmicrux/giltspine) dies, then the rest of its triggers do not resolve a la neffru. So if a cosmicrux damage somehow destroys giltspine or a second cosmicrux, the rest of those discards/damage are not dealt
- All of these damages are interleavable in APC order - eg you could do damage A, discard B, discard A, damage B

File: server/game/cards/14-DM/Bastionclaw.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ class Bastionclaw extends Card {
     // Your keys cost –1A for each forged key your opponent has.
     setupCardAbilities(ability) {
         this.persistentEffect({
-            targetController: 'self',
+            targetController: 'current',
```
Comment (dmikalova) [inline server/game/cards/14-DM/Bastionclaw.js:7]: This was lowering opponent key cost

File: server/game/cards/14-DM/ExeldonYash.js [hunk 7]
```diff
@@ -15,12 +15,15 @@ class ExeldonYash extends Card {
                 },
                 then: (preThenContext) => ({
                     condition: () =>
-                        !!preThenContext.target && preThenContext.source.hasToken('amber'),
-                    gameAction: ability.actions.returnAmber((context) => ({
-                        target: context.source,
+                        !!preThenContext.target &&
```
Comment (dmikalova) [inline server/game/cards/14-DM/ExeldonYash.js:18]: Similar to HP - Exeldon must capture exactly 2 amber and discard a card to if you do move an amber.

File: server/game/GameActions/PutIntoPlayAction.js [hunk 8]
```diff
@@ -36,10 +36,21 @@ class PutIntoPlayAction extends CardGameAction {
 
     // Gigantic creatures require 2 play allowances to play both halves.
     // Playing from hand always provides enough allowance.
-    canPutIntoPlayGigantic(context, card) {
+    canPutIntoPlayGigantic(_, card) {
         return card.location === 'hand' || this.numPlayAllowances >= 2;
     }
 
+    // Returns whether the other half of a gigantic creature has the given
```
Comment (dmikalova) [inline server/game/GameActions/PutIntoPlayAction.js:43]: Handles deploy for gigantics (monster zero) - whether from hand, discard, or put into play

File: server/game/gamesteps/forcedtriggeredabilitywindow.js [hunk 9]
```diff
@@ -86,8 +86,12 @@ class ForcedTriggeredAbilityWindow extends BaseStep {
             return true;
         }
 
-        let autoResolveChoice = this.choices.find((context) => context.ability.autoResolve);
-        if (autoResolveChoice) {
+        const autoResolveChoice = this.choices.find((context) => context.ability.autoResolve);
```
Comment (dmikalova) [inline server/game/gamesteps/forcedtriggeredabilitywindow.js:89]: Cosmicrux and Giltspine mesmerist should be orderable per creature readied, but also giltspine should autoresolve if no other abilities in same window.

I suspect this may fix certain other situations with people wanting something to be orderable, don't have an example though.

File: test/server/cards/06-WoE/MemroxTheRed.spec.js [hunk 10]
```diff
@@ -83,4 +83,45 @@ describe('Memrox the Red', function () {
             expect(this.player1.amber).toBe(4);
         });
     });
+
```
Comment (dmikalova) [inline test/server/cards/06-WoE/MemroxTheRed.spec.js:86]: While updating Exeldon/HP the question of whether Memrox properly handled discard from archive so I added validations for that

**Review by strib — APPROVED** [review]
